src/test/java/com/example/SpringExample/EmployeeServiceTests.java

```diff
@@ -0,0 +1,87 @@
+package com.example.SpringExample;
+
+import com.example.SpringExample.entity.Employee;
+import com.example.SpringExample.repository.EmployeeRepository;
+import com.example.SpringExample.service.EmployeeServiceImpl;
+import org.aspectj.lang.annotation.Before;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+import java.util.Optional;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class EmployeeServiceTests {
+
+    @Mock
+    private EmployeeRepository employeeRepository;
+
+    @InjectMocks
+    private EmployeeServiceImpl employeeService;
+
+    @Spy
+    private Employee employee;
+
+    @BeforeEach
+    public void setup(){
+        employee = new Employee.Builder()
+                .id(1L)
+                .firstName("Madhu")
+                .lastName("Shekhawat")
+                .email("madhu@gmail.com")
+                .build();
+    }
+
+//    @BeforeEach
+//    public void init(){
```

Comment:
Any reason for keeping this code snippet?

Reply:
The @ExtendWith and MockitoAnnotations.openMocks, are both used to initialize Mock. We can use any of them